src/apps/workspaces/service/user_applet_access.py

```diff
@@ -93,14 +93,18 @@ async def add_role_by_private_invitation(self, role: Role):
         owner_access = await UserAppletAccessCRUD(
             self.session
         ).get_applet_owner(self._applet_id)
+        user: User = await UsersCRUD(self.session).get_by_id(self._user_id)
         access_schema = await UserAppletAccessCRUD(self.session).save(
             UserAppletAccessSchema(
                 user_id=self._user_id,
                 applet_id=self._applet_id,
                 role=role,
                 owner_id=owner_access.user_id,
                 invitor_id=owner_access.user_id,
-                meta={},
+                meta=dict(
+                    secretUserId=str(uuid.uuid4()),
+                    nickname=f"{user.first_name} {user.last_name}",
+                ),
```

Comment:
Those fields are not actual for all roles, so should not be added without role checking, right?

Reply:
@vshvechko These fields are relevant only for the respondent. For other roles, the structure of the "Meta" field is different.

Reply:
@dparfeniuk so it needs to be changed, as you add those fields for all roles

Reply:
We use this code snippet only for inviting respondents. But I added a role check in case this fragment is used somewhere else.